wallet/multisig.go

```diff
@@ -185,3 +188,22 @@ func (w *Wallet) FetchAllRedeemScripts() ([][]byte, error) {
 	})
 	return redeemScripts, err
 }
+
+// PrepareRedeemMultiSigOutTxOutput estimates the tx value for a MultiSigOutTx
+// output and adds it
+func (w *Wallet) PrepareRedeemMultiSigOutTxOutput(msgTx *wire.MsgTx, p2shOutput *P2SHMultiSigOutput, pkScript *[]byte) error {
+	// estimate the output fee
+	txOut := wire.NewTxOut(0, *pkScript)
+	feeSize := txsizes.EstimateSerializeSize(len(msgTx.TxIn), []*wire.TxOut{txOut}, false)
+	feeEst := txrules.FeeForSerializeSize(w.RelayFee(), feeSize)
+	if feeEst >= p2shOutput.OutputAmount {
```

Comment:
i'm glad that the `internal` package forced to take another look at this code, because this is not doing what it should be.  there is no fee checking here.  what it should be checking is if the output is dust, and that can be checked using txrules.IsDustOutput.